dalesbred/src/main/java/org/dalesbred/query/SqlQuery.java

```diff
@@ -112,6 +120,47 @@ private SqlQuery(@NotNull @SQL String sql, @NotNull List<?> args) {
         return args;
     }
 
+
+    /**
+     * Returns the fetch size of this query
+     */
+    public @Nullable Integer getFetchSize() {
+        return fetchSize;
+    }
+
+    /**
+     * A non-null fetch size will be set as the fetch size for the statements executed from this query.
+     *
+     * @param size fetch size in rows or null
+     * @throws IllegalArgumentException if size is < 0
+     */
+    public void setFetchSize(@Nullable Integer size) {
+        if (size != null && size < 0)
+            throw new IllegalArgumentException("Illegal fetch size " + size + ". Fetch size must be null or > 0");
+        this.fetchSize = size;
+    }
+
+    /**
+     * Returns the fetch direction of ths query
+     */
+    public @Nullable Integer getFetchDirection() {
+        return fetchDirection;
+    }
+
+    /**
+     * A non-null fetch direction will be set as the fetch direction for the statements executed from this query.
+     * The direction must be null or one of {@link ResultSet#FETCH_FORWARD}, {@link ResultSet#FETCH_REVERSE} or {@link ResultSet#FETCH_UNKNOWN}.
+     *
+     * @param direction fetch direction or null
+     * @throws IllegalArgumentException if direction is not null or one of {@link ResultSet#FETCH_FORWARD}, {@link ResultSet#FETCH_REVERSE} or {@link ResultSet#FETCH_UNKNOWN}.
+     */
+    public void setFetchDirection(@Nullable Integer direction) {
+        if (direction != null &&
+                !(ResultSet.FETCH_FORWARD == direction || ResultSet.FETCH_REVERSE == direction || ResultSet.FETCH_UNKNOWN == direction))
```

Comment:
Having an enumeration will also simplify the implementation and documentation of this method by making illegal states statically impossible to represent.